### PR TITLE
Move checkpoint to boot.common, remove FEATURES checkpoint and getPhase(String), resalt ProtectedString

### PR DESCRIPTION
**Pull request by tjwatson**

### CODE REVIEW

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#libby

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
fixes #18607

**Comment by tjwatson** [comment]
@pgunapal please review commit 561fde6

**Review by pgunapal — APPROVED** [review]
The changes for re-salting the ProtectedString, looks good to me.

File: dev/com.ibm.ws.config/bnd.bnd [hunk 1]
```diff
@@ -95,4 +95,6 @@ instrument.classesExcludes: com/ibm/ws/config/internal/resources/*.class
 	com.ibm.ws.kernel.service;version=latest, \
 	com.ibm.ws.logging;version=latest, \
 	com.ibm.ws.kernel.boot.logging;version=latest, \
-	com.ibm.ws.kernel.security.thread;version=latest
+	com.ibm.ws.kernel.security.thread;version=latest,\
+	com.ibm.ws.kernel.boot.core;version=latest,\
```
Comment (sebratton) [inline dev/com.ibm.ws.config/bnd.bnd:99]: Did you need both core and common here?

**Review by sebratton — APPROVED** [review]

File: dev/com.ibm.ws.config/bnd.bnd (hunk 1 above)
Comment (tjwatson) [reply dev/com.ibm.ws.config/bnd.bnd:99]: > Did you need both core and common here?

This is the testpath.  The unit tests will fail without core on its classpath during execution.

**Comment by tjwatson** [comment]
#libby

**Pull request merged**